src/main/kotlin/com/workos/users/UsersApi.kt

```diff
@@ -735,4 +733,47 @@ class UsersApi(private val workos: WorkOS) {
     val config = RequestConfig.builder().data(options).build()
     return workos.post("/users/${options.userId}/send_verification_email", MagicAuthChallenge::class.java, config)
   }
+
+  @JsonInclude(JsonInclude.Include.NON_NULL)
+  class DeleteUserOptions @JvmOverloads constructor(
+    @JsonProperty("userId") val userId: String
+  ) {
+    init {
+      require(userId.isNotBlank()) { "User ID is required" }
+    }
+
+    /**
+     * Builder class for [DeleteUserOptions].
+     */
+    class DeleteUserOptionsBuilder {
+      private lateinit var userId: String
+
+      /**
+       * Sets the user ID.
+       */
+      fun userId(value: String) = apply { this.userId = value }
+
+      /**
+       * Creates a [DeleteUserOptions] with the given builder parameters.
+       */
+      fun build(): DeleteUserOptions {
+        return DeleteUserOptions(userId)
+      }
+    }
+
+    /**
+     * @suppress
+     */
+    companion object {
+      @JvmStatic
+      fun builder(): DeleteUserOptionsBuilder {
+        return DeleteUserOptionsBuilder()
+      }
+    }
+  }
+
+  fun deleteUser(deleteUserOptions: DeleteUserOptions) {
+    val id = deleteUserOptions.userId
+    workos.delete("/users/$id")
```

Comment:
Looks like the other delete functions in this SDK just take in the ID directly as a parameter:
```suggestion
  fun deleteUser(id: String) {
    workos.delete("/users/$id")
```

Reply:
I made this change, but fwiw, in node we made the decision to make these options rather than directly taking a string. I'm happy to keep the change or make the others the opts, just let me know what's preferred

Reply:
I think if we do that for this SDK we should probably do it to all of them and make a major version change.

Reply:
cool, major version changes are due anyways so can evaluate the